rfc/data-privacy/v2.md

```diff
@@ -0,0 +1,552 @@
+    
+    # Reusable transformers
+    transformers:
+      - name: "email"
+        expression: |
+          repeat("*", len(split(value, "@")[0])) + "@" + split(value, "@")[1]
+      - name: "default"
+        expression: |
+          valueDataType == "boolean" ? false : 
+          valueDataType == "number" ? 0 : 
+          repeat("*", len(value))
+      - name: "isoDate"
+        expression: |
+          date(value, "2006-01-02 15:04:05").Year() + "-**-**"
+    
+    # Obfuscation policies
+    policies:
+      - name: "Data Scientist Obfuscation"
+        # When this policy is active (evaluated once per request)
+        activate: "'data-scientist' in request.auth.roles"
+        
+        # YAML-based targeting (preferred for readability)
+        # Note: targets are evaluated first and take precedence over custom rules
+        targets:
+          - type: "User"
+            field: "email"
+            transform: "email"
+          - type: "User" 
+            field: "phoneNumber"
+            transform: "default"
+          - scalar_type: "ISODate"
+            transform: "isoDate"
+        
+        # For complex conditions not easily expressed in YAML
+        # Custom rule is only evaluated if no target rule matches
+        custom: |
+          # This expression is evaluated for each field
+          # It can access the field context and apply custom logic
+          typeName == "LogEntry" && fieldName == "ipAddress" ? 
+            replace(value, "\\d+$", "0") : 
+            value
+      
+      - name: "Developer Obfuscation"
+        activate: "'developer' in request.auth.roles"
+        # Note: Either use a default_transform OR custom, not both.
+        # When default_transform is specified, custom rules will be ignored.
+        default_transform: "default" # Applied to all fields not matched by targets
+        targets:
+          - type: "User"
+            field: "email"
+            transform: "email"
+      
+      - name: "Always Active Policy"
+        # A policy can be set to always be active without expression evaluation
+        always_active: true
+        targets:
+          - scalar_type: "EmailAddress"
+            transform: "email"
+```
+
+## Detailed Specifications
+
+### Policy Activation
+
+Policies can be activated in two ways:
+
+1. Using an `activate` expression that evaluates to `true` or `false` for each request:
+   ```yaml
+   activate: "'data-scientist' in request.auth.roles"
+   ```
+
+2. Using `always_active: true` for policies that should always be applied without expression evaluation:
+   ```yaml
+   always_active: true
+   ```
+
+> **Note:** `activate` and `always_active` are mutually exclusive. If both are specified, `always_active` takes precedence and the `activate` expression is ignored.
+
+### Target-Based Field Selection
+
+The `targets` field provides a readable, YAML-based way to specify which fields should be obfuscated and how:
```

Comment:
Would it make sense to change `field` to `fields` in case you want to apply the same obfuscation function to multiple fields in one type? That might omit a bit of redundant configuration

Reply:
we can, in fact, we can allow "fields" to be a selection set, so you can just define multiple space delimited fields

Reply:
Why space delimited and not a yaml array?

Reply:
yaml array is probably better 👍